packages/gatsby/src/utils/page-data.ts

```diff
@@ -127,9 +140,15 @@ export async function flush(): Promise<void> {
     // them, a page might not exist anymore щ（ﾟДﾟщ）
     // This is why we need this check
     if (page) {
+      const staticQueryHashes =
+        staticQueriesByTemplate.get(page.componentPath)?.map(String) || []
```

Comment:
We shouldn't map here, right? We should do it when we save identifiers.
```suggestion
        staticQueriesByTemplate.get(page.componentPath) || []
```